src/Orleans/Providers/ILeaseProvider.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Orleans.LeaseProviders
+{
+    /// <summary>
+    /// Aquired lease, which contains the resource key, duration of the lease and the token you can use to renew the lease with
+    /// </summary>
+    public class AcquiredLease
+    {
+        public string ResourceKey { get; }
+        public TimeSpan Duration { get; }
+        public string Token { get; }
+    }
+
+    /// <summary>
+    /// Lease request where you can specify ResourceKey, which you want to apply a lease on, and duration of your lease. 
+    /// </summary>
+    public class LeaseRequest
+    {
+        public string ResourceKey { get; set; }
+        public TimeSpan? Duration { get; set; }
+    }
+
+    /// <summary>
+    /// Lease provider interface 
+    /// </summary>
+    public interface ILeaseProvider
```

Comment:
>And LeaseProvider manages leases, so I think it is natural for it to have a query operation. Or did I overthink ?

Not necessarily. I think querying functionality will inevitably harm or complicate scalability. And debugging isn't a good justification for that to me. The extension interface path seems like a better approach. Even there I would be cautious about querying.